sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -262,9 +261,66 @@ static pi_result USMSetIndirectAccess(pi_kernel kernel) {
 
 extern "C" {
 
+// Helper functions
+
+// Returns true if the device is a cslice subdevice.
+static bool isCCS(pi_device device) {
+  if (!device)
+    return false;
+  return cslice_devices.find(device) != cslice_devices.end();
+}
+
+// Returns the underlying CL device.
+// For a regular pi_device, cl_device_id can be obtained by a simple typecast.
+// For a cslice subdevice, we explicitly store the cl_device_id and then
+// retrieve it when needed.
+static cl_device_id getClDevice(pi_device device) {
+  assert(device);
+  if (isCCS(device))
+    return cslice_devices[device].cl_dev;
+  else
+    return cast<cl_device_id>(device);
+}
+
+// Returns true if the device is a root device.
+static bool isRootDevice(pi_device device) {
+  if (!device)
+    return false;
+  cl_device_id parentId = nullptr;
+  clGetDeviceInfo(getClDevice(device), CL_DEVICE_PARENT_DEVICE,
+                  sizeof(cl_device_id), &parentId, NULL);
+  if (parentId == nullptr)
+    return true;
+  return false;
+}
+
+// Returns the list of underlying cl_devices.
+static std::vector<cl_device_id> getClDevices(pi_uint32 num_devices,
+                                              const pi_device *devices) {
+  std::vector<cl_device_id> cl_devices(num_devices);
+  for (size_t i = 0; i < num_devices; ++i)
+    cl_devices[i] = getClDevice(devices[i]);
+  return cl_devices;
+}
+
+// Return true if the device is a PVC device
+static bool isPVC(pi_device device) {
+  // Identify device name.
+  const size_t MAXDEVICENAMELEN = 64;
+  std::string device_name(MAXDEVICENAMELEN, ' ');
+  cl_int res = clGetDeviceInfo(getClDevice(device), CL_DEVICE_NAME,
+                               MAXDEVICENAMELEN, &device_name[0], nullptr);
+  return (res == CL_SUCCESS) &&
+         (device_name.find("0x0bd5") != std::string::npos);
```

Comment:
Is this sufficient for checking for Intel GPUs? 

Thanks

Reply:
Since it looks like this is checking for a PVC device by looking for the PCI device ID, I'd recommend querying this directly vs. parsing the device name.  You can query for the PCI device ID by querying CL_DEVICE_ID_INTEL from [cl_intel_device_attribute_query](https://registry.khronos.org/OpenCL/extensions/intel/cl_intel_device_attribute_query.html#_modifications_to_the_opencl_api_specification).

Reply:
Modified the check in upcoming commit. Thanks